### DISCUSSION TITLE
Error 404 Handling During Debugging

### DISCUSSION THREAD

**Discussion by rmcsqrd** [nginxinc/nginx-s3-gateway#37] [Ideas]
Hi there, thanks for making this - I got it working successfully and it is very useful.

I ran into issues with the REST authentication because MFA was enabled for the AWS account tied to my access key/secret key pair and ended up needing to make a new role with reduced permissions and no MFA (obviously this is not at all related to your package).

During the debugging process I found the the default 404 response behavior was a bit of a red herring in the debugging process (but understand its utility for production systems). I spent a lot of time trying a lot of different things before I dug into the actual nginx conf file and found the part of the configuration that [changed the error response](https://github.com/nginxinc/nginx-s3-gateway/blob/master/common/etc/nginx/templates/default.conf.template#L107-L112). Once I commented these lines out locally I saw I was actually getting a 403 Access Denied error which helped narrow down the debugging process. 

I think it would be useful to mention this within the "Getting Started Guide" as a potential way to debug any authentication error. I also think it might be useful to mention the MFA thing as well. I am happy to file a documentation PR calling this out if you think it would be useful.

**Comment by rmcsqrd** [comment]
Hi @dekobon - thanks for the quick response. I am not suggesting any added functionality to the actual project - just suggesting an addendum to the "Getting Started Guide" documentation for ways I found useful to "debug" the 403 error response from S3. 

I personally would have found useful some documentation about the 404 error code redirect behavior of the nginx configuration when identifying why I couldn't get it to work. The actual nginx configuration file was well documented but it took a little bit of digging to find. Admittedly maybe this is something most people would do right off the bat but it was non-obvious to me at the time. 

Again - totally up to you if you would like me to submit a documentation only PR; information on the default 404 error behavior would have been personally useful to me but am also cognizant I am only one data point.

**Reply by dekobon** [reply]
Oh, @rmcsqrd , somehow I got my wires crossed and replied to the wrong discussion thread. My above answer was for someone else. I'll reply here in a moment.

**Reply by dekobon** [reply]
Yes, adding additional information to the documentation about how to turn off the 404 error codes upon failure would be really useful for folks trying to debug. It is a manner of working that I've done without much thought, but I'm sure is not intuitive to anyone other than myself. A comment in the configuration may also be useful.

We would much appreciate a PR to improve the experience.
